Add Style/IndentAssignment Cop

panthomakos opened the pull request:
This is a prequel to #2361. It is enabled by default and properly fixes the auto-correct output that `MultilineAssignmentLayout` produces.

When assignments span multiple lines, this cop enforces proper indentation on the right-hand-side of the expression. It only corrects the first line, as the remaining lines can be corrected by other cops such as `Style/IndentationConsistency`.

    # bad
    value =
    if foo
      'bar'
    else
      'biz'
    end

    # good
    value =
      if foo # corrected/accepted
      'bar'
    else
      'biz'
    end

I have also added a fix to `Astrolabe::Node#asgn_method_call?`. It used to consider `!=` method calls as assignments - which they are not. There were also no tests for this method so I've added them.

jonas054 commented on spec/rubocop/ast_node_spec.rb:
```diff
@@ -0,0 +1,25 @@
+require 'spec_helper'
+
+describe Astrolabe::Node do
+  context '#asgn_method_call?' do
```
I think this ought to be `describe`, not `context`.

panthomakos commented:
@jonas054 updated to address your comment

jonas054 commented:
The code looks good, but there's something wrong. The build fails. Please add spec example(s) that fails the same way that the rubocop self-inspection currently does (if you see what I mean), and fix.

panthomakos commented:
@jonas054 The issue was some additional comparison operators that `ast_node` was counting as assignment operators (such as `===`). I've adjusted the list and added some additional tests. I've also added a backup test to `indent_assignment_spec.rb` to ensure that missing operator calls don't cause failures.

jonas054 commented:
:+1: Looks good but you must rebase on top of the latest `master`.

panthomakos commented:
@jonas054 rebased on top of `master`

bbatsov commented on lib/rubocop/ast_node.rb:
```diff
@@ -37,8 +37,11 @@ def single_line?
       !multiline?
     end
 
+    COMPARISON_OPERATORS = [:==, :===, :!=, :<=, :>=, :>, :<, :<=>]
```
This should be frozen and possibly placed higher up the class def. I don't like mixing constants with method definitions.

bbatsov commented:
I don't see a test showing that this won't generate an offense:

```ruby
ala = if bala
           tralala
         end
```

bbatsov commented:
> I have also added a fix to Astrolabe::Node#asgn_method_call?. It used to consider != method calls as assignments - which they are not. There were also no tests for this method so I've added them.

Nice!

alexdowad commented:
> I have also added a fix to Astrolabe::Node#asgn_method_call?

Thanks for fixing my bugs!
